code/modules/mob/living/silicon/robot/robot_defense.dm

```diff
@@ -320,9 +320,15 @@ GLOBAL_LIST_INIT(blacklisted_borg_hats, typecacheof(list( //Hats that don't real
 	if(. & EMP_PROTECT_SELF)
 		return
 	switch(severity)
-		if(1)
+		if(EMP_HEAVY)
 			Stun(160)
-		if(2)
+			if(prob(70) && !shell) // AI shell is ignored
+				lawupdate = FALSE
+				set_connected_ai(null)
+				src.add_ion_law(generate_ion_law())
```

Comment:
dont do src.proc

Reply:
i know the old code does it but thats no excuse to repeat its mistakes, would be better if you removed it from there